targets/TARGET_STM/TARGET_STM32L4/flash_api.c

```diff
@@ -137,6 +137,9 @@ int32_t flash_erase_sector(flash_t *obj, uint32_t address)
         return -1;
     }
 
+    /* Clear error programming flags */
+    __HAL_FLASH_CLEAR_FLAG(FLASH_FLAG_ALL_ERRORS);
```

Comment:
If you clear all flash error, maybe you can remove next __HAL_FLASH_CLEAR_FLAG ?

Reply:
Should be OK now. FLASH_FLAG_ALL_ERRORS includes also FLASH_FLAG_OPTVERR